app/app.go

```diff
@@ -964,8 +964,8 @@ func (app *App) ProcessTxConcurrent(
 ) {
 	defer wg.Done()
 	// Store the Channels in the Context Object for each transaction
-	ctx.WithTxBlockingChannels(getChannelsFromSignalMapping(txBlockingSignalsMap))
-	ctx.WithTxCompletionChannels(getChannelsFromSignalMapping(txCompletionSignalingMap))
+	ctx = ctx.WithTxBlockingChannels(getChannelsFromSignalMapping(txBlockingSignalsMap))
+	ctx = ctx.WithTxCompletionChannels(getChannelsFromSignalMapping(txCompletionSignalingMap))
```

Comment:
Doesn't fix it e2e but a bad enough bug to get into the branch sooner so I don't forget.

ctx is immutable